File: Sources/APTradeDomain/Position.swift
import Foundation

public struct Position: Equatable, Codable, Sendable {
    public let asset: Asset
    public let quantity: Quantity
    public let averageCost: Money
    public let realizedPnL: Money

    public init(asset: Asset, quantity: Quantity, averageCost: Money, realizedPnL: Money) {
        self.asset = asset
        self.quantity = quantity
        self.averageCost = averageCost
        self.realizedPnL = realizedPnL
    }

    public func marketValue(at price: Money) -> Money {
        Money(amount: price.amount * quantity.amount, currencyCode: price.currencyCode)
    }

    public func unrealizedPnL(at price: Money) -> Money {
        Money(amount: (price.amount - averageCost.amount) * quantity.amount,
              currencyCode: price.currencyCode)
    }
}

File: shared/src/commonMain/kotlin/com/aptrade/shared/domain/Position.kt
package com.aptrade.shared.domain

import com.ionspin.kotlin.bignum.decimal.BigDecimal

data class Position(
    val asset: Asset,
    val quantity: BigDecimal,
    val averageCost: Money,
    val realizedPnL: Money,
) {
    fun marketValue(at: Money): Money =
        Money(at.amount * quantity, at.currencyCode)

    fun unrealizedPnL(at: Money): Money =
        Money((at.amount - averageCost.amount) * quantity, at.currencyCode)
}
